auth/src/adapter/redis-database-adapter.ts

```diff
@@ -1,12 +1,12 @@
-import Redis from 'redis';
+import Redis from 'ioredis';
 
 import { Database } from '../api/interfaces/database';
 import { Logger } from '../api/services/logger';
 
 export class RedisDatabaseAdapter extends Database {
     private readonly redisPort = parseInt(process.env.CACHE_PORT || '', 10) || 6379;
     private readonly redisHost = process.env.CACHE_HOST || '';
-    private database: Redis.RedisClient;
+    private database: Redis.Redis;
```

Comment:
I'll put a comment here, but it belongs to the complete file:

You are using a set to manage all sessions, this is better than using `keys`. But this is still not performant when using many sessions. I mean you are trying to map userId<->SessionId, right? And you want to access both directions efficiently. My assumption here: A user has multiple sessions, but a sessions belongs to exactly one user. So you can use the following setup in Redis:

Use a hash for sessionId->userId, e.g. with the key `auth:session:sessionIds`. If you have a session Id, you can use `HGET`, `HEXISTS`, `HDEL` to manage it and `HSET` to add new ones. For the other relation sets (which you are using) are great. Make a set per user, e.g. use the key `auth:session:user:<id>` with a set of sessionIds. One remark: Instead of using `SINTER`, you can use `SMEMBERS`.

With this setup you have to manage session- and userIds always twice, but you can do every lookup in your code very performant instead of iterating over all sessions.

Another thing: You are using boolean status codes, e.g. for `set`. But as far as I can tell, it is never evaluated (e.g. in the session service). Whats about using Errors for errorhandling? Set should have a void return code. If not, throw an error. Than you can include the error message within the error.

I still have additional questions we should discuss later.